### PR TITLE
GoDocs for completion, dashboard and diagnostics cli commands

### PR DESCRIPTION
**Pull request by ArthurSens**
Added comments to document several methods and structs on cmd package. Based on GoDoc guidelines.

Comments added to `completion`, `dashboard` and `diagnostics` cli commands

Signed-off-by: arthursens arthursens2005@gmail.com

### CODE REVIEW

**Comment by cpretzer** [comment]
@ArthurSens I'm looking through these changes now and want to make sure that you saw the DCO check: https://github.com/linkerd/linkerd2/blob/master/CONTRIBUTING.md#developer-certificate-of-origin

Please be sure to amend your commit to include the DCO and thanks for help!

**Comment by ArthurSens** [comment]
yeah @cpretzer , I'm aware of the DCO... I've messed up somehow when doing the commits and I still couldn't fix it.

I've asked on the contributors channel on slack how to solve this, @olix0r is trying to help me there! Hopefully I'll fix it soon

**Comment by cpretzer** [comment]
hi @ArthurSens would you mind taking a look at the Go formatting issue? https://github.com/linkerd/linkerd2/pull/4518/checks?check_run_id=728943041

Definitely want to get these changes merged, then we can start looking at some other `good first issues`! 😄

**Comment by ArthurSens** [comment]
Oh, I'm sorry it took me so long to fix this. The file that failed was totally unrelated to the changes I've made, so I thought it was a CI error. 

I should have asked about this a long time ago

**Comment by cpretzer** [comment]
thanks @ArthurSens looks like there are some conflicts with the main branch. Do you mind rebasing and force pushing a commit so that there are no conflicts?

**Comment by cpretzer** [comment]
@ArthurSens Just following up on my slack message:

you can rebase using the `--signoff` flag to get the DCO

It is unexpected that the rebase left the file in a state of conflict. Might be necessary to manually resolve the conflict

File: cli/cmd/diagnostics.go [hunk 1]
```diff
@@ -14,16 +14,23 @@ const (
...
 )
 
+// diagnosticsOptions holds values for command line flags that apply to the dashboard
+// command. All fields in this struct should have corresponding flags added in
+// the newDiagnosticsOptions func later in this file.
 type diagnosticsOptions struct {
 	wait time.Duration
 }
 
+// newDiagnosticsOptions initializes dashboard options with default wait time.
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply cli/cmd/diagnostics.go:24]: Sounds reasonable.

Should I describe the command options documented on that [previous PR](https://github.com/linkerd/linkerd2/pull/4412) too?

[the remark this answers is not in the document]
Comment (cpretzer) [reply cli/cmd/diagnostics.go:24]: I'd think of it in terms of "what will be helpful to someone reading the comments?"

So, if someone reading the code might wonder what those options are for, then it's useful information. On the other hand, if the usage of the options is obvious from the code, then they probably don't need to be documented.

File: cli/cmd/alpha_clients.go [hunk 2]
```diff
@@ -12,10 +12,15 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// alphaClientsOptions holds values for command line flags that apply to the alpha
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply cli/cmd/alpha_clients.go]: I don't really know why these changes got into this PR... Should I do something to remove them?

File: cli/cmd/dashboard.go [hunk 3]
```diff
@@ -35,13 +35,20 @@ const (
 	defaultPort = 50750
 )
 
+// dashboardOptions holds values for command line flags that apply to the dashboard
+// command. All fields in this struct should have corresponding flags added in
+// the newCmdDashboard func later in this file.
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply cli/cmd/dashboard.go:40]: To be honest I copied these comments from a [similar method](https://github.com/linkerd/linkerd2/blob/main/cli/cmd/install.go#L35) from install.go file. 

I believe that it is saying that all option should have a value when the struct is created, but I agree that this part is not necessary and I will remove it

File: cli/cmd/diagnostics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArthurSens) [reply cli/cmd/diagnostics.go:24]: For this file, in particular, almost all options are properly documented [here](https://github.com/linkerd/linkerd2/blob/main/cli/cmd/dashboard.go#L15-L36)

It doesn't seem to be the case for other cli functions though, I'll keep an eye on it from now on

**Pull request merged**